packages/react-core/src/components/ContextSelector/ContextSelectorToggle.tsx

```diff
@@ -66,7 +66,7 @@ export class ContextSelectorToggle extends React.Component<ContextSelectorToggle
   onEscPress = (event: any) => {
     const { isOpen, parentRef, onToggle } = this.props;
     const keyCode = event.keyCode || event.which;
-    if (isOpen && keyCode === KEY_CODES.ESCAPE_KEY && parentRef && parentRef.contains(event.target)) {
+    if (isOpen && keyCode === KEY_CODES.ESCAPE_KEY && parentRef?.current && !parentRef.current.contains(event.target)) {
```

Comment:
```suggestion
    if (isOpen && keyCode === KEY_CODES.ESCAPE_KEY) {
```
This will allow the toggle to be closed with the Escape key.

Reply:
right, that makes sense, I don't see why the reference was required in the first place